### DISCUSSION TITLE
[1장]퀴즈

### DISCUSSION THREAD

**Discussion by hnitam** [FE-tech-talk/TechTalk-React#5] [Q&A]
- 즉시 실행 함수(IIFE)의 사용 목적?장점?
- 화살표 함수와 일반 함수에서 this 바인딩 차이점?
- 이벤트 루프 역할?

**Comment by easyhyun00** [comment]
## 답변

1. 함수를 정의하고 그 순간 즉시 실행되어 한 번만 호출되고 다시 호출할 수 없음 그래서 재사용되지 않고, 단 한 번만 실행할 경우 사용하면 좋다

**Comment by hnitam** [comment]
- 즉시 실행 함수(IIFE)의 사용 목적?장점?
    - 글로벌 스코프를 오염시키지 않는 독립적인 함수 스코프를 운용할 수 있다.
    - 코드를 읽는 이에게 이 함수는 다시 호출되지 않는다는 점을 각인시켜 리팩터링에 도움이 된다.
- 화살표 함수와 일반 함수에서 this 바인딩 차이점?
    - 일반 함수 : this가 함수를 정의할 때 결정되는 것이 아니라, 함수가 어떻게 호출되느냐에 따라 동적으로 결정된다. 내부의 this는 전역 객체를 가리킨다.
    - 화살표 함수 : this가 함수를 선언하는 시점에 결정된다. 함수 자체의 바인딩을 갖지 않는다. 화살표 함수 내부에서 상위 스코프의 this를 그대로 따른다. 
- 이벤트 루프 역할?
    - 호출 스택에 실행 중인 코드가 있는 지, 태스크 큐에 대기 중인 함수가 있는 지 반복해서 확인하는 역할을 한다. 
    - 호출 스택이 비었다면 태스크 큐에 대기 중인 작업이 있는지 확인하고, 이 작업을 실행 가능한 오래된 것부터 순차적으로 꺼내와서 실행한다. 태스크 큐가 빌 때까지 한다.
